Misc/name-me.sh

```diff
@@ -0,0 +1,357 @@
+			unset funcname
+			return 0
+		elif [ "$DEBUG" = 1 ]; then
+			$PRINTF "$FIXME_FORMAT_STRING_DEBUG" "$2"
+			$PRINTF "$FIXME_FORMAT_STRING_DEBUG_LOG" "$2" >> "$logPath"
+			unset funcname
+			return 0
+		else
+			die 255 "processing DEBUG variable with value '$DEBUG' in $funcname"
+		fi
+	else
+		die 255 "processing variable IGNORE_FIXME with value '$IGNORE_FIXME' in $0"
+	fi
+}; alias efixme='efixme "$LINENO"'
+
+# Resolve root
+edebug "Resolving root on user with ID '$(id -u)"
+if [ "$(id -u)" = 0 ]; then
+	edebug "Script has been executed as user with ID 0, assuming root"
+	# NOTICE(Krey): We are prefixing root commands with '$SUDO', this is done to make sure that we are not using sudo here
+	unset SUDO
+# NOTICE(Krey): The ID 33333 is used by gitpod
+elif [ "$(id -u)" = 1000 ] || [ "$(id -u)" = 33333 ]; then
+	ewarn "Script $myName is not expected to run as non-root, trying to elevate root.."
+	if command -v sudo 1>/dev/null; then
+		einfo "Found 'sudo' that can be used for root elevation"
+		SUDO=sudo
+	elif command -v su 1>/dev/null; then
+		einfo "Found 'su' that can be used for a root elevation"
+		ewarn "This will require the end-user to parse a root password multiple times assuming that root has a password set"
+		SUDO=su
+	elif ! command -v sudo 1>/dev/null && ! command -v su 1>/dev/null; then
+		die 3 "Script $myName depends on root permission to install packages where commands 'sudo' nor 'su' are available for root elevation"
+	else
+		die 225 "processing root on non-root"
+	fi
+else
+	die 3 "Unknown user ID '$(id -u)' has been parsed in script $myName"
+fi
+
+# Identify system and core
+edebug "Resolving Kernel used"
+if command -v $UNAME 1>/dev/null; then
+	KERNEL="$($UNAME -s)"
+
+	case "$KERNEL" in
+		Linux)
+			edebug "Identified kernel as '$KERNEL'"
+			# Identify distribution
+			if command -v lsb_release 1>/dev/null; then
+				# NOTICE(Krey): Command 'lsb_release -si' returns values alike 'Debian' where logic expects 'debian' -> piped in tr to standardize
+				DISTRO="$(lsb_release -si | tr '[:upper:]' '[:lower:]')"
+			elif [ -f /etc/os-release ]; then
+				DISTRO="$(grep -o "^ID\=.*" /etc/os-release | sed 's#^ID\=##gm')"
+			elif ! command -v lsb_release 1>/dev/null && [ ! -f /etc/os-release ]; then
+				die 1 "Unable to identify distribution since command 'lsb_release' and file /etc/os-release are not present"
+			else
+				die 255 "identifying distro in $myName running logic for Linux"
+			fi
+
+			# FIXME(Krey): Better logic needed
+			if command -v lsb_release 1>/dev/null; then
+				RELEASE="$(lsb_release -cs)"
+			elif ! command -v lsb_release 1>/dev/null; then
+				ewarn "Unable to identify distribution using command 'lsb_release' since it is not present"
+			else
+				die 255 "Identifying distribution"
+			fi
+
+			# Install dependencies
+			case "$DISTRO" in debian|ubuntu)
+				efixme "Do not run 'apt-get update' if it's not needed"
+			esac
+			case "$DISTRO/$RELEASE" in
```

Comment:
Note that this is where the dependencies can be specified per kernel, distro and distro release ^-^